lenskit-core/src/main/java/org/lenskit/util/math/RollingWindowMeanAccumulator.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2016 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.util.math;
+
+import java.util.Arrays;
+
+/**
+ * Accumulates the mean of the most recent accumulated values. Meaning that its reported mean will only reflect the N
+ * most recent values. This is accomplished using a fixed size array and a running total. Adding and getting the average
+ * are take O(1).
+ *
+ * @author <a href="http://www.grouplens.org">GroupLens Research</a>
+ */
+public class RollingWindowMeanAccumulator {
+    private static final int DEFAULT_SIZE = 100;
+
+    private final double[] data;
+    // invariant - sum of data.
+    private double sum;
+    // invariant - if position is negative then the next index is data.length + position (and we are filling the data
+    // array for the first time) otherwise position is the next index (meaning the index of the oldest value).
+    private int position;
```

Comment:
Interesting design here - not quite how I would have done it, but that probably doesn't matter.

I would use two variables, `next` and `size`, with the update rule `next = (next + 1) % size`.